lib/app/admin/data-reset/reset-users.js

```diff
@@ -0,0 +1,27 @@
+var Promise = require('bluebird');
+var $fh = require('fh-mbaas-api');
+
+var authServiceGuid = process.env.WFM_AUTH_GUID;
+
+module.exports = function() {
+
+  return Promise.fromCallback(function(fromCallbackCb) {
+    $fh.service({
+      guid: authServiceGuid,
+      path: "/admin/data",
+      method: "DELETE",
+      timeout: 25000
+    }, function(err, body, res) {
+      //resolve manually. $fh.service returns error on internal (fh) or connection error.
+      // we can check statusCode and body for service side errors - code != 204
+      if (err) {
+        fromCallbackCb(err);
+      } else if (res && res.statusCode !== 204) {
+        fromCallbackCb(body);
```

Comment:
Nope. There's no custom error handling for fh-mbaas-api -- it's just a wrapper around request(), which only gives an err if it has trouble getting a response from the server. If the server responds, with any status code, then there's no error, and it's the user's responsibility to catch it. (Fun fact -- by default, the `request-promise` library *will* throw an error for a 4xx or 5xx status code, and you can opt-out with an option.) 

There's no requirement for the server to send any body back with any response, so you shouldn't make that assumption. Also, though FH code is notorious for not doing this, you really should be putting an Error instance in the callback, not just some arbitrary (maybe null) object.

Another funny thing I just thought about: a common server response is 201 (created) with no message body. In that case, the response would go through this error path, but because the body was empty, it wouldn't look like an error an all.